src/atomate2/vasp/jobs/phonons.py

```diff
@@ -0,0 +1,447 @@
+                force_diagonal=kwargs["force_diagonal"],
+                force_90_degrees=False,
+            )
+            transformation.apply_transformation(structure=structure)
+
+    supercell_matrix = transformation.transformation_matrix.tolist()
+    return supercell_matrix
+
+
+@job
+def generate_phonon_displacements(
+    structure: Structure,
+    supercell_matrix: np.array,
+    displacement: float,
+    sym_reduce: bool,
+    symprec: float,
+    use_symmetrized_structure: Union[str, None],
+    kpath_scheme: str,
+    code: str,
+):
+    """
+    will generate displacement structures based on phonopy
+    Parameters
+    ----------
+    structure: Structure object
+    supercell_matrix: np.array
+        array to describe supercell matrix
+    displacement: float
+        displacement in Angstrom
+    sym_reduce: bool
+        if True, symmetry will be used to generate displacements
+    symprec: float
+        precision to determine symmetry
+    use_symmetrized_structure: str»|None
+        primitive, conventional or None
+    kpath_scheme: str
+        scheme to generate kpath
+    code:
+        code to perform the computations
+    """
+    cell = get_phonopy_structure(structure)
+    if code == "vasp":
+        factor = VaspToTHz
+    # a bit of code repetition here as I currently
+    # do not see how to pass the phonopy object?
+    if use_symmetrized_structure == "primitive" and kpath_scheme != "seekpath":
+        primitive_matrix: Union[List[List[float]], str] = [
+            [1.0, 0.0, 0.0],
+            [0.0, 1.0, 0.0],
+            [0.0, 0.0, 1.0],
+        ]
+    else:
+        primitive_matrix = "auto"
+    phonon = Phonopy(
+        cell,
+        supercell_matrix,
+        primitive_matrix=primitive_matrix,
+        factor=factor,
+        symprec=symprec,
+        is_symmetry=sym_reduce,
+    )
+    phonon.generate_displacements(distance=displacement)
+
+    supercells = phonon.supercells_with_displacements
+
+    displacements = []
+    for cell in supercells:
+        displacements.append(get_pmg_structure(cell))
+    return displacements
+
+
+@job(output_schema=PhononBSDOSDoc, data=[PhononDos, PhononBandStructureSymmLine])
+def generate_frequencies_eigenvectors(
+    structure: Structure,
+    supercell_matrix: np.array,
+    displacement: float,
+    sym_reduce: bool,
+    symprec: float,
+    use_symmetrized_structure: Union[str, None],
+    kpath_scheme: str,
+    code: str,
+    displacement_data: dict[str, list],
+    total_energy: float,
+    epsilon_static: Matrix3D = None,
+    born: Matrix3D = None,
+    born_run_job_dir: str | Path | None = None,
+    born_run_uuid=None,
+    static_run_job_dir: str | Path | None = None,
+    static_run_uuid=None,
+    optimization_run_job_dir: str | Path | None = None,
+    optimization_run_uuid=None,
+    # combine serval of these options
+    npoints_band: int = 100,
+    kpoint_density_dos: int = 7000,
+    tol_imaginary_modes: float = 1e-5,
+    tmin=0,
```

Comment:
Please see other comment about moving these optional arguments into **kwargs.

Reply:
Yes, will do. I am just sometimes a bit worried that people don't see through these **kwargs as it complicates usage in my opinion. Not sure everyone immediately understands the Structure.from_file(...) in pymatgen for example.

Reply:
Yes, I also have this worry. Many people do not understand it. Perhaps you could add an explanation in the docstring pointing people to this other method?

Also, as this is a common practice in atomate2, we could even have a dedicated page in the docs about how to use atomate2 effectively, with some common design principles.

Reply:
Both suggestions sound like a great idea. 

I was also thinking about whether there should be a GUI at some point for people to build larger workflows by drag and drop or similar. You might have thought about it already and it would also be a massive amount of work ... but it's really hard to get into the usage workflows at the moment. Or we add even more tutorials...

Reply:
I think in the short term more tutorials is definitely needed. I was even thinking about adding a tutorial for most substantial workflows. E.g., we could add a workflow on how to do phonons using atomate2. It is more work from the developer side but it would definitely increase adoption!

As for a GUI, I know @mkhorton has a great vision for this, and its also something I'd really want to do. I'm not sure how to make it happen, specially regarding funding for the project. Realistically, it will need an experienced web developer working for ~1 year minimum. 

I actually think this could be something which a spin-off company would be well suited for. Create a web application that could be licensed out to industry to fund the development (academia could be free/get a discount). But I'm aware this doesn't fit into the ethos of MP.

Reply:
I agree that this will be challenging. Maybe, we could also think about a template for developing your own workflow and mention the common steps in workflow development with important classes/methods for this? This is how I would start implementing something like this with a new co-worker as well. I don't think that we can expect typical chemists/physicits to navigate new and complex codes very well. 

I am really happy about the test documentation.

Reply:
@davidwaroquiers I would love to join the discussion as it actually fits one of my broader and long-term interests.